README.md

```diff
@@ -15,6 +18,101 @@ def deps do
 end
 ```
 
+## Configuration
+
+To configure the library use the `config/config.exs` to define `host`,
+`client_id` and `client_secret`. It will end up looking something like:
+
+```elixir
+  config :my_app, :gamora,
+    identity_provider: [
+      host: "https://oidc.com",
+      client_id: "MY_APP_CLIENT_ID",
+      client_secret: "MY_APP_CLIENT_SECRET",
+      token_path: "/oauth/token", # This is the default
+      introspect_path: "/oauth/introspect" # This is the default
+    ]
+```
+
+## Usage
+
+### Protected Routes
+
+Use the plug `Gamora.Plugs.AuthenticatedUser` in your protected routes.
+This will get the access token from cookies and validate it against
+the IDP (OIDC Identity Provider).
+
+```elixir
+defmodule MyAppWeb.Router do
+  use MyAppWeb, :router
+
+  # ... pipelines
+
+  pipeline :protected do
+    plug Gamora.Plugs.AuthenticatedUser,
+      callbacks: MyAppWeb.GamoraCallbacks
+  end
```

Comment:
if the config could be set it as an option instead of a global config, this will allow to have different pipelines with different providers, supporting multiple oauth servers with the same library, I have seen this more and more in libraries where the glboal config is changed in favor of something like or in plug options or in GenServer init arguments

Reply:
I understand what you said, but it is not common that applications like have more than one OIDC Relying Party. Usuallly you have just one client_id and client_secret. I mean it is not impossible to have more than one but it is not common.

If I do what you suggested, we should pass the same options in multiple parts of the router, for the plug that validates the access token, for the endpoint that is going to generate the access token using an authorization code and for the callback endpoint when the app is web-based.

I undertstand what you said but IMO this OIDC client implementation is not a good candidate to do it.
What do you think?